enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -986,7 +986,10 @@ func NewSchedulerServerWithOptions(env environment.Env, options *Options) (*Sche
 	return s, nil
 }
 
-func (s *SchedulerServer) GetPoolInfo(ctx context.Context, os, requestedPool, workflowID string, useSelfHosted bool) (*interfaces.PoolInfo, error) {
+func (s *SchedulerServer) GetPoolInfo(ctx context.Context, os, requestedPool, workflowID string, useSelfHosted *bool) (*interfaces.PoolInfo, error) {
```

Comment:
is a *bool better than making an enum type and passing that here?

I think you could define an enum like this:
```
const (
  selfHostedPoolRequested = iota
  sharedPoolRequested
)
```

then parse your platform property into that enum and pass it.

Reply:
I prefer `*bool` over a custom enum type since `use-self-hosted-executors` is an optional bool (unset/nil, false, or true).

I could see `*bool` being considered a little unusual though so I threw an enum into `interfaces.go`, lmk what you think